File: C/0009.c
#include <stdio.h>

int main() {
    int a, b, c;
    int target_sum = 1000;

    // Iterate over possible values of a and b
    for (a = 1; a < target_sum / 3; a++) {
        for (b = a + 1; b < target_sum / 2; b++) {
            c = target_sum - a - b;
            // Check if it forms a Pythagorean triplet
            if (a * a + b * b == c * c) {
                printf("The Pythagorean triplet is: a = %d, b = %d, c = %d\n", a, b, c);
                printf("The product abc is: %d\n", a * b * c);
                return 0;
            }
        }
    }

    printf("No Pythagorean triplet found.\n");
    return 1;
}

File: TypeScript/0009.ts
function findPythagoreanTripletProduct(sum: number): number {
    for (let a = 1; a < sum / 3; a++) {
        for (let b = a + 1; b < (sum - a) / 2; b++) {
            const c = sum - a - b;

            // Check if it's a Pythagorean triplet
            if (a * a + b * b === c * c) {
                return a * b * c; // Return the product of the triplet
            }
        }
    }

    return -1; // If no triplet is found
}

// Usage
const sum = 1000;
console.log(`The product of the Pythagorean triplet for which a + b + c = ${sum} is:`, findPythagoreanTripletProduct(sum));